app/jobs/v3/service_instance_async_job.rb

```diff
@@ -0,0 +1,174 @@
+require 'jobs/reoccurring_job'
+
+module VCAP::CloudController
+  module V3
+    class ServiceInstanceAsyncJob < VCAP::CloudController::Jobs::ReoccurringJob
+      attr_reader :warnings
+
+      def initialize(guid, audit_info)
+        super()
+        @service_instance_guid = guid
+        @user_audit_info = audit_info
+        @warnings = []
+        @request_attr = {}
+        @first_time = true
+      end
+
+      def perform
+        client = VCAP::Services::ServiceClientProvider.provide({ instance: service_instance })
+
+        check_pre_conditions
+
+        begin
+          if @first_time
+            compute_maximum_duration
+            execute_request(client)
+            compatibility_checks
+            @first_time = false
+          end
+
+          gone! && return if service_instance.blank?
```

Comment:
what does `return` do here?

Reply:
It makes sure we exit the method. `gone!` raises on provision and update, but not on deprovision